x/retry/retry.go

```diff
@@ -57,19 +62,30 @@ func WithPolicyProvider(provider PolicyProvider) MiddlewareOption {
 	})
 }
 
+// TallyScope sets a Tally scope that will be used to record retry metrics.
+func TallyScope(scope tally.Scope) MiddlewareOption {
```

Comment:
Naming: Just Metrics or Scope?

Consistency: At least one other option in this package for the same type uses `WithFoo(..)` while this is just `Foo(..)`.